internal/infrastructure/logger/gorm_log.go

```diff
@@ -2,92 +2,73 @@ package logger
 
 import (
 	"context"
-	"fmt"
-	"os"
+	"errors"
 	"time"
 
-	"github.com/podengo-project/idmsvc-backend/internal/config"
-	"github.com/rs/zerolog"
+	"golang.org/x/exp/slog"
+	"gorm.io/gorm"
 	"gorm.io/gorm/logger"
 )
 
-type gormZerolog struct {
-	logger zerolog.Logger
+type gormLogger struct {
+	slogger              *slog.Logger
+	ignoreRecordNotFound bool
 }
 
-func NewGormLog(cfg *config.Config) logger.Interface {
-	var o *os.File
-	if cfg.Logging.Console {
-		o = os.Stdout
-	} else {
-		o = os.Stderr
-	}
-
-	lvl, err := zerolog.ParseLevel(cfg.Logging.Level)
-	if err != nil {
-		panic(err)
-	}
-	l := zerolog.New(o).
-		Level(lvl).
-		With().
-		Timestamp().
-		Logger()
-
-	return &gormZerolog{
-		logger: l,
-	}
+func (l *gormLogger) LogMode(level logger.LogLevel) logger.Interface {
+	return l
 }
 
-func (l *gormZerolog) LogMode(level logger.LogLevel) logger.Interface {
-	switch level {
-	case logger.Silent:
-		{
-			l.logger.Level(zerolog.NoLevel)
-		}
-	case logger.Info:
-		{
-			l.logger.Level(zerolog.InfoLevel)
-		}
-	case logger.Warn:
-		{
-			l.logger.Level(zerolog.WarnLevel)
-		}
-	case logger.Error:
-		{
-			l.logger.Level(zerolog.ErrorLevel)
-		}
-	}
-	return l
+func (l *gormLogger) Info(ctx context.Context, msg string, args ...interface{}) {
+	l.slogger.InfoCtx(ctx, msg, args)
 }
 
-func (l *gormZerolog) Info(ctx context.Context, msg string, args ...interface{}) {
-	if l.logger.GetLevel() <= zerolog.InfoLevel {
-		l.logger.Info().Msgf(msg, args...)
-	}
+func (l *gormLogger) Warn(ctx context.Context, msg string, args ...interface{}) {
+	l.slogger.WarnCtx(ctx, msg, args)
 }
-func (l *gormZerolog) Warn(ctx context.Context, msg string, args ...interface{}) {
-	if l.logger.GetLevel() <= zerolog.WarnLevel {
-		l.logger.Warn().Msgf(msg, args...)
-	}
+
+func (l *gormLogger) Error(ctx context.Context, msg string, args ...interface{}) {
+	l.slogger.ErrorCtx(ctx, msg, args)
 }
-func (l *gormZerolog) Error(ctx context.Context, msg string, args ...interface{}) {
-	if l.logger.GetLevel() <= zerolog.ErrorLevel {
-		l.logger.Error().Msgf(msg, args...)
+
+func (l *gormLogger) Trace(
+	ctx context.Context,
+	begin time.Time,
+	fc func() (sql string, rowsAffected int64),
+	err error,
+) {
+	elapsedTime := time.Since(begin)
+
+	if err != nil && (!errors.Is(err, gorm.ErrRecordNotFound) ||
+		!l.ignoreRecordNotFound) {
+		sql, rows := fc()
+
+		l.slogger.Log(
+			ctx,
+			LevelTrace,
+			err.Error(),
```

Comment:
This confused me for a sec. `err.Error()` returns a string, which is then used a the primary message in the logger call.

Reply:
Well, what should be put as the message instead?